javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java

```diff
@@ -1238,18 +1241,24 @@ private void doExtract(FileExtractor extractor, Path file, ExtractorState state)
       for (ParseError err : errors) {
         String msg = "A parse error occurred: " + StringUtil.quoteWithBackticks(err.getMessage().trim())
             + ". Check the syntax of the file. If the file is invalid, correct the error or [exclude](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/customizing-code-scanning) the file from analysis.";
-        // file, relative to the source root
-        DiagnosticLocation.Builder builder = DiagnosticLocation.builder();
+
+        Optional<DiagnosticLocation> diagLoc = Optional.empty();
         if (file.startsWith(LGTM_SRC)) {
-          builder = builder.setFile(file.subpath(LGTM_SRC.getNameCount(), file.getNameCount()).toString());
-        }
-        DiagnosticLocation diagLoc = builder
+          diagLoc = DiagnosticLocation.builder()
+            .setFile(file.subpath(LGTM_SRC.getNameCount(), file.getNameCount()).toString()) // file, relative to the source root
```

Comment:
Taking the relative path with `LGTM_SRC` is not always right. `LGTM_SRC` is simply the value of the current work directory and may not actually be the "repository root", for example when a custom `source-root` is used or the working directory is changed.

Reply:
Thanks — this seems like a fairly ingrained assumption in the autobuilder, so in the interests of breaking work down, I'll leave this for now as it is, and open an issue on the JS repo with your comments.